static/app/views/releases/thresholdsList/index.tsx

```diff
@@ -48,51 +55,58 @@ function ReleaseThresholdList({}: Props) {
     selectedEnvs: selection.environments,
   });
 
-  // const _getAllSelectedProjects = (): Project[] => {
-  //   return projects.filter(project =>
-  //     selection.projects.some(id => String(id) === project.id || id === -1)
-  //   );
-  // };
-
-  // const _getAllEnvironments = (): string[] => {
-  //   const selectedProjects = selection.projects.map(id => String(id));
-  //   const {user} = ConfigStore.getState();
-  //   const allEnvSet = new Set(projects.flatMap(project => project.environments));
-  //   // NOTE: mostly taken from environmentSelector.tsx
-  //   const unSortedEnvs = new Set(
-  //     projects.flatMap(project => {
-  //       /**
-  //        * Include environments from:
-  //        * all projects if the user is a superuser
-  //        * the requested projects
-  //        * all member projects if 'my projects' (empty list) is selected.
-  //        * all projects if -1 is the only selected project.
-  //        */
-  //       if (
-  //         (selectedProjects.length === 1 &&
-  //           selectedProjects[0] === ALL_ACCESS_PROJECTS &&
-  //           project.hasAccess) ||
-  //         (selectedProjects.length === 0 && (project.isMember || user.isSuperuser)) ||
-  //         selectedProjects.includes(project.id)
-  //       ) {
-  //         return project.environments;
-  //       }
-
-  //       return [];
-  //     })
-  //   );
-  //   const envDiff = new Set([...allEnvSet].filter(x => !unSortedEnvs.has(x)));
-
-  //   return Array.from(unSortedEnvs)
-  //     .sort()
-  //     .concat([...envDiff].sort());
-  // };
-
-  // NOTE: currently no way to filter for 'None' environments
+  const selectedProjects: Project[] = useMemo(() => {
+    return projects.filter(project =>
+      selection.projects.some(id => String(id) === project.id || id === -1)
+    );
+  }, [projects, selection.projects]);
+
+  const getAllEnvironments = (): string[] => {
+    const selectedProjectIds = selection.projects.map(id => String(id));
+    const {user} = ConfigStore.getState();
+    const allEnvSet = new Set(projects.flatMap(project => project.environments));
+    // NOTE: mostly taken from environmentSelector.tsx
+    const unSortedEnvs = new Set(
+      projects.flatMap(project => {
+        /**
+         * Include environments from:
+         * all projects if the user is a superuser
+         * the requested projects
+         * all member projects if 'my projects' (empty list) is selected.
+         * all projects if -1 is the only selected project.
+         */
+        if (
+          (selectedProjectIds.length === 1 &&
+            selectedProjectIds[0] === String(ALL_ACCESS_PROJECTS) &&
+            project.hasAccess) ||
+          (selectedProjectIds.length === 0 && (project.isMember || user.isSuperuser)) ||
+          selectedProjectIds.includes(project.id)
```

Comment:
Would it be better to put these into constants? It's a little difficult to tell what's what, even with the comment explanation above

Reply:
hmmm i'm not sure if sticking these into a constant would make it anymore clear either :/